src/main/java/edu/cursor/hw12/service/UserService.java

```diff
@@ -0,0 +1,41 @@
+package edu.cursor.hw12.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.cursor.hw12.entities.User;
+import edu.cursor.hw12.handler.UserExceptionHandler;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.HashMap;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserExceptionHandler userExceptionHandler;
```

Comment:
you don't need to inject `UserExceptionHandler`
if an exception will occur - ControllerAdvice automatically intercept an exception and process it

Reply:
fixed